src/goal.cpp

```diff
@@ -14,22 +14,29 @@
 
 namespace pick_ik {
 
+auto linear_distance(Eigen::Isometry3d frame_1, Eigen::Isometry3d frame_2) {
+    return (frame_1.translation() - frame_2.translation()).norm();
+}
+
+auto angular_distance(Eigen::Isometry3d frame_1, Eigen::Isometry3d frame_2) {
+    auto const q_1 = Eigen::Quaterniond(frame_1.rotation());
+    auto const q_2 = Eigen::Quaterniond(frame_2.rotation());
+    return q_2.angularDistance(q_1);
+}
```

Comment:
Input args should be `Eigen::Isometry3d const&` for both of these.

Also, we already know the output there is `double`, so we can directly set that istead of `auto`.